### PR TITLE
Double similarity function - null pointer exception

### PR DESCRIPTION
**Pull request by navinrathore**
* fixed config.json as MatchType NUMERIC_WITH_UNITS changed to NUMBER_WITH_UNITS
* 'null' checks in DoubleSimilarityFunction. Also, checked for Double.NAN

### CODE REVIEW

**Comment by navinrathore** [comment]
Tested with febrl standard dataset and another updated with Double field/values
with amazon-google dataset
with iTunes-amazon dataset

**Comment by navinrathore** [comment]
Noticed that, for amazon-google dataset, header for google.csv file is not compatible with names in schema. Though, zingg seems working fine but below is the message.
```
2022-06-28 11:39:15,950 [main] WARN  zingg.util.PipeUtil - Reading Pipe [name=google, format=CSV, preprocessors=null, props={location=examples/amazon-google/GoogleProducts.csv, delimiter=,, header=true}, schema=StructType(StructField(id,StringType,false), StructField(title,StringType,true), StructField(description,StringType,true), StructField(manufacturer,StringType,true), StructField(price,DoubleType,true))]
 2022-06-28 11:39:16,248 [Executor task launch worker for task 1.0 in stage 10.0 (TID 39)] WARN  org.apache.spark.sql.catalyst.csv.CSVHeaderChecker - CSV header does not conform to the schema.
 Header: id, name, description, manufacturer, price
 Schema: id, title, description, manufacturer, price
Expected: title but found: name
CSV file: file:///home/navin/workDir/zingg-1/examples/amazon-google/GoogleProducts.csv

```

**Comment by navinrathore** [comment]
* #351

File: core/src/test/java/zingg/similarity/function/TestDoubleSimilarityFunction.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+public class TestDoubleSimilarityFunction {
+	
+	
+	@Test
+	public void testFirstNumIsNull() {
+		DoubleSimilarityFunction isNull = new DoubleSimilarityFunction();
```
Comment (sonalgoyal) [inline core/src/test/java/zingg/similarity/function/TestDoubleSimilarityFunction.java:12]: name of variable?

**Review by sonalgoyal — CHANGES_REQUESTED** [review]
- Add tests for some doubles.
- Rename variable

File: core/src/test/java/zingg/similarity/function/TestDoubleSimilarityFunction.java (hunk 1 above)
Comment (navinrathore) [reply core/src/test/java/zingg/similarity/function/TestDoubleSimilarityFunction.java:12]: changed to simFn

Comment (navinrathore) [reply core/src/test/java/zingg/similarity/function/TestDoubleSimilarityFunction.java:12]: Added some tests

**Pull request merged**